### PR TITLE
Add c/c++ header icons by changing c/c++ icons colors to purple 

### PR DESCRIPTION
**Pull request by Caraczan**
Ideally it would be completely different icon. But this will be much better then current icons.

### CODE REVIEW

**Comment by TheronLeshkojL** [comment]
It seems like a commit message from a GitHub repository. The first commit appears to add and build C/C++ headers while changing the default icon to a purple CL. The second commit seems to be a merge of the previous pull request.

**Comment by Caraczan** [comment]
@TheronLeshkojL exactly, wasn't sure how to do pull request not mine project, so I forked project, created different branch and after commiting changes merged into master and then created pull request.

I've duplicated icons and config according to how its done, and then changed color to purple. Maybe I will change icons later, but it  was annoying me that .cpp and .hpp/.h were the same color.

**Comment by Caraczan** [comment]
Hey @braver, I've created icons for .h and .hpp files (basicly `H` and `H++`). But after convertion their respective .pngs are black instead of purple. 
Then I converted them to path and joined them into one path, but it still doesn't color. 

What can I do to make them up to standard ? 

I've created them in inkscape and exported to standard `svg`.

They are not in this pull.

File: build/assets/file_type_cheader.svg [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     inkscape:window-height="2087"
+     inkscape:window-x="0"
+     inkscape:window-y="0"
+     inkscape:window-maximized="1"
+     inkscape:current-layer="svg7" />
+  <defs
+     id="defs4">
+    <style
+       id="style2">
+      .i-color {
```
Comment (braver) [inline build/assets/file_type_cheader.svg:37]: this `i-color` class isn't applied to the path element, which is probably why the color also doesn't change

File: build/assets/file_type_cppheader.svg [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+     inkscape:zoom="64.1875"
+     inkscape:cx="5.8266796"
+     inkscape:cy="8.0311587"
+     inkscape:window-width="1672"
+     inkscape:window-height="1233"
+     inkscape:window-x="0"
+     inkscape:window-y="0"
+     inkscape:window-maximized="0"
+     inkscape:current-layer="svg7" /><defs
+     id="defs4"><style
```
Comment (braver) [inline build/assets/file_type_cppheader.svg:36]: duplicate style element

File: build/assets/file_type_cheader.svg [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<svg
+   width="16"
+   height="16"
+   viewBox="0 0 16 16"
+   version="1.1"
+   id="svg7"
+   sodipodi:docname="file_type_cheader.svg"
+   inkscape:version="1.2.2 (b0a8486541, 2022-12-01)"
```
Comment (braver) [inline build/assets/file_type_cheader.svg:9]: can Inkscape maybe export a clean svg without all this useless (for us) metadata in it?

Comment (Caraczan) [reply build/assets/file_type_cheader.svg]: Okay I searched and found there is `optimized svg`, which removed all those misc information.

File: build/assets/file_type_cppheader.svg (hunk 2 above)
Comment (Caraczan) [reply build/assets/file_type_cppheader.svg]: removed duplicate style

File: build/assets/file_type_cheader.svg (hunk 1 above)
Comment (Caraczan) [reply build/assets/file_type_cheader.svg:5]: added class="i-color" to paths and color is applied now. :)

**Comment by Caraczan** [comment]
The only concern for me is weird graphical bug/"i dont know how to create proper svgs" where on places of connection that has existed before is weird blur/border.

**Comment by Caraczan** [comment]
@braver should I resolve individual conversations, or is it your task there ?

**Pull request closed without merging**